articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+    $taskResource = Invoke-RestMethod `
+        -Method GET `
+        -Uri "https://management.azure.com$($scenarioTask.ProjectTask.Id)?api-version=2018-03-31-preview&`$expand=output" `
+        -ContentType "application/json" `
+        -Authentication Bearer `
+        -Token $token;
+    
+    $taskResource.properties;
+}
+
+# create the get table task properties by initializing the connection and 
+# database name
+$getTablesTaskProperties = @{
+    "input"    = @{
+        "connectionInfo"    = $null;
+        "selectedDatabases" = $null;
+    };
+    "taskType" = "GetUserTablesMySql";
+};
+
+LogMessage -Message "Running scenario to get the list of tables from the target database..." -IsProcessing $true
+$getTablesTaskProperties.input.connectionInfo = $targetConnInfo;
+$getTablesTaskProperties.input.selectedDatabases = @($TargetDatabaseName);
+# Create a name for the task
+$getTableTaskName = "$($TargetDatabaseName)GetUserTables"
+# Get the list of tables from the source
+$getTargetTablesTask = RunScenario -MigrationService $dmsService `
+    -MigrationProject $dmsProject `
+    -ScenarioTaskName $getTableTaskName `
+    -TaskProperties $getTablesTaskProperties;
+
+if (-not ($getTargetTablesTask)) { throw "ERROR: Could not get target database $TargetDatabaseName table information." }
+LogMessage -Message "List of tables from the target database acquired."
+
+LogMessage -Message "Running scenario to get the list of tables from the source database..." -IsProcessing $true
+$getTablesTaskProperties.input.connectionInfo = $sourceConnInfo;
+$getTablesTaskProperties.input.selectedDatabases = @($SourceDatabaseName);
+# Create a name for the task
+$getTableTaskName = "$($SourceDatabaseName)GetUserTables"
+# Get the list of tables from the source
+$getSourceTablesTask = RunScenario -MigrationService $dmsService `
+    -MigrationProject $dmsProject `
+    -ScenarioTaskName $getTableTaskName `
+    -TaskProperties $getTablesTaskProperties;
+
+if (-not ($getSourceTablesTask)) { throw "ERROR: Could not get source database $SourceDatabaseName table information." }
+LogMessage -Message "List of tables from the source database acquired."
+
+```
+
+## Step 8: Build table mapping based on user configuration
+
+As part of configuring the migration task, you will create a mapping between the source and target tables. The mapping is at the table name level but the assumption is that the table structure (column count, column names, data types etc.) of the mapped tables is exactly the same.
+
+The following script creates a mapping based on the target and source table list extracted in **Step 7**. For partial data load, the user can provide a list of table to filter out the tables. If no user input is provided, then all target tables are mapped. The script also checks if a table with the same name exists in the source or not. If table name does not exists in the source, then the target table is ignored for migration. 
```

Comment:
I kind of think the reverse, if no input is provided, I think we should assume that we will migrate all tables from the source.  If they want to migrate a subset of the tables, they should use the -SelectedTables option to filter out tables they don't want to migrate.  I think customers would rather see a failure by default if migrating from a source to a target and the target is missing a subset of the tables.  To me that is better than silently migrating a subset of the data from the source.  If a subset is to be migrated, I think it should be an intentional selection passed into the script.

Reply:
I understand your point of view and as a result i have changed the approach:
1. Added two lists IncludeList and ExcludeList
2. IncludeList - Default $null => Include all source tables
3. ExcludeList - Default $null => Exclude no source tables
4. ExcludeList has higher priority than IncludeList
5. Check if table in exclude list - if not
6. Check if table exists in target - if not log Table_Mapping_Error and continue to next table
7. Add to mapping if either include list is null or table in include list
8. If any Table_Mapping_Error found then stop process

Please take a look at the new implementation and i think this will be more aligned to your thought